lib/Context/Internal/ProcessHub.php

```diff
@@ -37,7 +40,11 @@ public function __construct()
         if ($isWindows) {
             $this->uri = "tcp://127.0.0.1:0";
         } else {
-            $this->uri = "unix://" . \tempnam(\sys_get_temp_dir(), "amp-parallel-ipc-") . ".sock";
+            $temp = \tempnam(\sys_get_temp_dir(), "amp-parallel-ipc-");
+            unlink($temp);
```

Comment:
We should simply avoid `tempnam` here, we don't have any advantage of using it here.

Reply:
What's the purpose of the `unlink` call before adding `'.sock'` to the name?

Reply:
Tempnam creates a temporary file, it doesn't simply give you a name.

Reply:
I didn't realize that. I would avoid `tempnam` too and use a few random bytes instead to generate a name.